### ISSUE TITLE
Same concept appears multiple times in search results

### ISSUE THREAD

**Issue by osma** [NatLibFi/Skosmos#53]
## At which URL did you encounter the problem?
http://dev.finto.fi/sv/search?lang=&q=ufologia&vocabs=yso

## What steps will reproduce the problem?
1. Go to above URL (or perform another search which matches both the prefLabel and altLabel of a concept)

## What is the expected output? What do you see instead?

Expected to see the concept only once, but instead seeing several matches: one via prefLabel, another via altLabel. These should probably be merged into just one visible result, but how to do that is a difficult question. The SPARQL queryConcepts method considers all matches separately.

**Comment by mariaforsen** [comment]
http://finto.fi/yso/sv/search?lang=&q=k%C3%A4sittely 

Very confusing way to show the results like above. The same concept should be shown only once. Thus when I search with "käsittely" the resultlist should show only: käsittely (keskustelu ja päättäminen) (fi), käsittely (kooste) ja käsittely (toisenlaiseksi muuttaminen) (fi) and the short list should also show the concept in other languages.

**Comment by osma** [comment]
We'll see if it can be easily done for 1.0, it's a difficult problem to fix

**Comment by osma** [comment]
Nothing to be done in 1.0. Will have to reconsider after #190 is complete, as it will anyways affect a lot of things.

**Comment by osma** [comment]
#190 was done. This issue is still a problem in a few edge cases, e.g.
http://finto.fi/yso/sv/search/?clang=sv&anylang=on&q=k%C3%A4sittely
http://finto.fi/yso/sv/search/?clang=sv&anylang=on&q=ufologia

**Comment by osma** [comment]
Possibly affected by #273

**Comment by osma** [comment]
Will try to fix this as part of #273

**Comment by osma** [comment]
With the recent refactoring of generateConceptSearchQuery into an outer and inner query, this should be possible to implement by editing the inner query. But because of #392, it is difficult to make sure that the best possible explanations of the match will be shown.

**Comment by osma** [comment]
We need two variants of the inner queries. One which returns only unique concepts, another which returns all matches (as currently). This should be controlled by a parameter `unique` which should be exposed also to the REST API `search` method.

The autocomplete functionality should show all matches. The actual search result list should be restricted to unique matches.

**Comment by osma** [comment]
REST search queries with no language parameters don't return results if unique=true.
http://localhost/Skosmos/rest/v1/yso/search?query=aateli*&unique=1

**Comment by osma** [comment]
Now I think we only need documentation about the REST API change, and a unit test

**Comment by osma** [comment]
REST API page in wiki updated.

**Comment by osma** [comment]
Need to check that the order of result makes sense. It should be based on the matched label.

**Comment by osma** [comment]
Still problems with some wildcard queries when using GenericSparql, for example `*kyrkok*`

**Comment by osma** [comment]
All should be fixed now, just need a unit test
